tests/shallow.test.tsx

```diff
@@ -3,11 +3,12 @@ import { act, fireEvent, render } from '@testing-library/react'
 import { beforeEach, describe, expect, it, vi } from 'vitest'
 import { create } from 'zustand'
 import { useShallow } from 'zustand/react/shallow'
+import { createWithEqualityFn } from 'zustand/traditional'
```

Comment:
This change naively assumes that replacing `createWithEqualityFn` is meant to be tested here. 

Is `createWithEqualityFn` going to be deprecated in v5 and dropped in v6?

Reply:
> This change naively assumes that replacing `createWithEqualityFn` is meant to be tested here.

Looks good. I think we should have done that before. In that sense, we can make this change directly in to `main`.

> Is `createWithEqualityFn` going to be deprecated in v5 and dropped in v6?

We haven't decided nor discussed. That's one possibility though.

Reply:
> Looks good. I think we should have done that before. In that sense, we can make this change directly in to main.

I'll make that change.

> We haven't decided nor discussed. That's one possibility though.

I made a discussion here: https://github.com/pmndrs/zustand/discussions/2243

Reply:
I made a PR to update the tests in `main`: https://github.com/pmndrs/zustand/pull/2244